src/eradiate/radprops/absorption.py

```diff
@@ -3,163 +3,774 @@
+    # select the bin
+    kw = ds.sigma_a.sel(
+        w=wm,
+        method="nearest",
+    ).expand_dims("w")
+
+    # 2. interpolation along g-point (pseudo-spectral coordinate)
+    return kw.interp(g=g).drop_vars("g")
+
+
+def eval_sigma_a_ckd_impl_ds(
+    ds: xr.Dataset,
+    thermoprops: xr.Dataset,
+    w: pint.Quantity,
+    g: float,
+    error_handler_config: dict[str, dict[str, str]] | None = None,
+) -> pint.Quantity:
+    """
+    Evaluate absorption coefficient in given spectral and thermophysical
+    conditions.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Absorption coefficient dataset.
+
+    thermoprops : Dataset
+        Atmospheric thermophysical properties data set.
+
+    w : quantity
+        Wavelength.
+
+    g: float
+        g-point.
+
+    error_handler_config : dict
+        Error handler configuration.
 
     Returns
     -------
     quantity
         Absorption coefficient values.
-
-    Raises
-    ------
-    ValueError
-        When wavelength, pressure, or temperature values are out of the range
-        of the data set and the corresponding fill value in ``fill_values`` is
-        ``None``.
-
-    Warnings
-    --------
-    The values of the absorption cross-section at the desired wavelength,
-    pressure and temperature values,
-    :math:`\\sigma_{a\\lambda} (p, T)`,
-    are obtained by interpolating the input absorption cross-section data
-    set along the corresponding dimensions.
-
-    Notes
-    -----
-    The absorption coefficient is given by:
-
-    .. math::
-        k_{a\\lambda} = n \\, \\sigma_{a\\lambda} (p, T)
-
-    where
-
-    * :math:`k_{a\\lambda}` is the absorption coefficient [:math:`L^{-1}`],
-    * :math:`\\lambda` is the wavelength [:math:`L`],
-    * :math:`n` is the number density [:math:`L^{-3}`],
-    * :math:`\\sigma_a` is the absorption cross section [:math:`L^2`],
-    * :math:`p` is the pressure [:math:`ML^{-1}T^{-2}`] and
-    * :math:`t` is the temperature [:math:`\\Theta`].
     """
+    if error_handler_config is None:
+        error_handler_config = DEFAULT_HANDLER_CONFIG
 
-    if fill_values is None:
-        fill_values = dict(w=None, pt=None)
-
-    if methods is None:
-        methods = dict(w="linear", pt="linear")
-
-    for name in ["w", "pt"]:
-        if name not in fill_values:
-            fill_values[name] = None
-        if name not in methods:
-            methods[name] = "linear"
-
-    # Interpolate along wavenumber dimension
-    xsw = ds.interp(
-        w=(1.0 / wl).m_as(ds.w.units),  # wavenumber in cm^-1
-        method=methods["w"],
-        kwargs=dict(
-            bounds_error=(fill_values["w"] is None),
-            fill_value=fill_values["w"],
-        ),
-    )
+    kw = interp_wg(ds, w, g)
+    kwx = interp_x(kw, thermoprops, error_handler_config["x"])
+    kwxp = interp_p(kwx, thermoprops, error_handler_config["p"])
+    kwxpt = interp_t(kwxp, thermoprops, error_handler_config["t"])
```

Comment:
Maybe expand `kw` to a more explicit name?

Reply:
I find it explicit enough, but feel free to rename it to your liking :)